### ISSUE TITLE
[Feature] add callback API for Autogluon's model.fit()

### ISSUE THREAD

**Issue by MountPOTATO** [autogluon/autogluon#2637]
While I am using Autogluon as a part of my labeling-training workflow, I find there are improvements that can be made for Autogluon to support better integrations with user-customized workflow. Sample code might be something like this, which is to add callback API during model fitting.

```python
# callback function(s), called during epochs
model = MultiModalPredictor(label='label').fit(train_data,
                                               callback=my_callback_function())
```

A similar feature is implemented by Keras (docs at https://keras.io/api/callbacks/). Some of the callbacks supported by Keras have been built in autogluon's `model.fit()` , and it seems reasonable if users want some customize callback to be performed. An example is the Keras `RemoteMonitor` (docs at https://keras.io/api/callbacks/remote_monitor/). Such feature allows a frontend (displaying the training process charts) to be updated constantly when the backend (running  `fit()` ) sends the data (epoch loss,  acc, etc.) during epochs. 

I believe feature like this can help Autogluon to display the training process information to a hosted WebUI frontend or Tensorboard in real-time. The frontend application here can be self-implemented (like Microsoft's NNI  framework, who launchs a dashboard WebUI when automl searching is conducted), which might take lots of effort. It can also be provided by user, with Autogluon simply serves as the sender of training data through a callback API.
